### PR TITLE
Fixes for `MethodInstance` objects that do not come from a Method

### PR DESCRIPTION
**Pull request by staticfloat**
When running `descend()` on a MethodInstance that does not come from a `Method`, we need to avoid attempting to tie back to the original source or method definition names.  This is the case when you have a chunk of IR that you have generated, but no real method to map back to it with.

### CODE REVIEW

**Review by timholy — APPROVED** [review]

**Comment by timholy** [comment]
Any chance of a test?

File: test/test_Cthulhu.jl [hunk 1]
```diff
@@ -973,4 +973,28 @@ end
     end
 end
 
+@testset "Bare-bones MIs" begin
```
Comment (aviatesk) [inline test/test_Cthulhu.jl:976]:
```suggestion
@static VERSION ≥ v"1.9-" && @testset "Bare-bones MIs" begin
```

**Review by aviatesk — APPROVED** [review]

File: src/Cthulhu.jl [hunk 2]
```diff
@@ -511,7 +511,7 @@ function _descend(term::AbstractTerminal, interp::AbstractInterpreter, curs::Abs
                     ioctx = IOContext(iostream,
                         :color => true,
                         :displaysize => displaysize(iostream), # displaysize doesn't propagate otherwise
-                        :SOURCE_SLOTNAMES => Base.sourceinfo_slotnames(codeinf),
+                        :SOURCE_SLOTNAMES => codeinf === nothing ? false : Base.sourceinfo_slotnames(codeinf),
```
Comment (aviatesk) [inline src/Cthulhu.jl:514]: Is this change necessary? `codeinf` here should always be `CodeInfo` object.

Comment (staticfloat) [reply src/Cthulhu.jl:514]: We did need it, yes, I'll send you a MWE.
